### DISCUSSION TITLE
Set _id from another input field

### DISCUSSION THREAD

**Discussion by AaronPorts** [Automattic/mongoose#14310] [Q&A] [answered]
Is it possible to create a setter for the `_id` field to take a value from another input field (eg `id` ) when saving a new model? Something like `Model.create({ id: a, key: value })` -> will create document `{ _id: a, key: value }` .

**Comment by vkarpov15** [comment, marked as answer]
Yes, however you would have to add one yourself using something like the following:

```
schema.virtual('id').get(function() { return this._id; }).set(function(v){ this._id = v; return v; });
```

Mongoose briefly had a virtual like this between versions 7.4 and 8.0, but we had to remove it because the virtual caused too many hard-to-debug issues for existing codebases to enable it by default. But there's nothing stopping you from adding that to your codebase.
